### ISSUE TITLE
Degradation in 1.32: Error "Cannot rootify root"

### ISSUE THREAD

**Issue by pyhedgehog** [arkane-systems/genie#100]
In previous version:
```console
root@honeypot-wsl:~# genie --version
1.31
root@honeypot-wsl:~# logout
root@honeypot-wsl:~# id
uid=0(root) gid=0(root) groups=0(root)
root@honeypot-wsl:~# echo ${INSIDE_GENIE:-false}
false
root@honeypot-wsl:~# genie -s
root@honeypot-wsl:~# echo ${INSIDE_GENIE:-false}
true
```
After upgrade:
```console
root@honeypot-wsl:~# genie --version
1.32
root@honeypot-wsl:~# genie -s
Unhandled exception: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.InvalidOperationException: Cannot rootify root.
   at ArkaneSystems.WindowsSubsystemForLinux.Genie.RootPrivilege..ctor()
   at ArkaneSystems.WindowsSubsystemForLinux.Genie.Program.ShellHandler(Boolean verbose)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Delegate.DynamicInvokeImpl(Object[] args)
   at System.Delegate.DynamicInvoke(Object[] args)
   at System.CommandLine.Invocation.ModelBindingCommandHandler.InvokeAsync(InvocationContext context)
   at System.CommandLine.Invocation.InvocationPipeline.<>c__DisplayClass4_0.<<BuildInvocationChain>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseParseErrorReporting>b__21_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass16_0.<<UseHelp>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass25_0.<<UseVersionOption>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass23_0.<<UseTypoCorrections>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseSuggestDirective>b__22_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseParseDirective>b__20_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseDebugDirective>b__11_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<RegisterWithDotnetSuggest>b__10_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass14_0.<<UseExceptionHandler>b__0>d.MoveNext()
```

**Comment by cerebrate** [comment]
Oops, my bad. (Although I wouldn't have expected it to work in 1.31-, either, due to not anticipating anyone running genie as root - real root, not suid root - in the first place.)

Will try and get fix out soonest.

**Comment by pyhedgehog** [comment]
Don't hurry. I have more bugs to report. ;-)

**Comment by aeryxium** [comment]
I'm getting this in 1.33 without running as root on Arch with WSL2.

`PS C:\Users\aeryx> wsl -l -v
  NAME         STATE           VERSION
* ArchLinux    Running         2
PS C:\Users\aeryx> wsl -u aeryxium -d ArchLinux genie -s
Unhandled exception: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.InvalidOperationException: Cannot rootify root unless REALLY root.
   at ArkaneSystems.WindowsSubsystemForLinux.Genie.RootPrivilege..ctor()
   at ArkaneSystems.WindowsSubsystemForLinux.Genie.Program.ShellHandler(Boolean verbose)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Delegate.DynamicInvokeImpl(Object[] args)
   at System.Delegate.DynamicInvoke(Object[] args)
   at System.CommandLine.Invocation.ModelBindingCommandHandler.InvokeAsync(InvocationContext context)
   at System.CommandLine.Invocation.InvocationPipeline.<>c__DisplayClass4_0.<<BuildInvocationChain>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseParseErrorReporting>b__21_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass16_0.<<UseHelp>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass25_0.<<UseVersionOption>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass23_0.<<UseTypoCorrections>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseSuggestDirective>b__22_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseParseDirective>b__20_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseDebugDirective>b__11_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<RegisterWithDotnetSuggest>b__10_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass14_0.<<UseExceptionHandler>b__0>d.MoveNext()
PS C:\Users\aeryx> wsl -u aeryxium -d ArchLinux
% id
uid=1001(aeryxium) gid=1001(aeryxium) groups=1001(aeryxium),998(wheel)
% genie --version
1.33`

But running as root works fine:

`PS C:\Users\aeryx> wsl -d ArchLinux genie -s
Waiting for systemd....!!!
# id
uid=0(root) gid=0(root) groups=0(root)
# genie --version
1.33
# echo ${INSIDE_GENIE:-false}
true`

**Comment by cerebrate** [comment]
Should be fixed as of 1.33.

**Comment by aeryxium** [comment]
Indeed fixed for me, thanks!
